Assets/LeapMotionModules/Hands/Scripts/RiggedHand.cs

```diff
@@ -76,25 +106,101 @@ public class RiggedHand : HandModel {
       Vector3 forward = trs.zBasis.ToVector3();
       return Quaternion.LookRotation(forward, up);
     }
-
+    /**Sets up the rigged hand by finding base of each finger by name */
     [ContextMenu("Setup Rigged Hand")]
     public void SetupRiggedHand() {
       modelFingerPointing = new Vector3(0, 0, 0);
       modelPalmFacing = new Vector3(0, 0, 0);
-      findFingerModels();
-      modelPalmFacing = calculateModelPalmFacing();
+      assignRiggedFingersByName();
+      SetupRiggedFingers();
+      modelPalmFacing = calculateModelPalmFacing(palm, fingers[1].transform, fingers[2].transform);
       modelFingerPointing = calculateModelFingerPointing();
       setFingerPalmFacing();
     }
+    /**Sets up the rigged hand if RiggedFinger scripts have already been assigned using Mecanim values.*/
+    public void AutoRigRiggedHand(Transform palm, Transform finger1, Transform finger2) {
+      Debug.Log("AutoRigRiggedHand()");
+      modelFingerPointing = new Vector3(0, 0, 0);
+      modelPalmFacing = new Vector3(0, 0, 0);
+      SetupRiggedFingers();
+      modelPalmFacing = calculateModelPalmFacing(palm, finger1, finger2);
+      modelFingerPointing = calculateModelFingerPointing();
+      setFingerPalmFacing();
+    }
+    /**Finds palm and finds root of each finger by name and assigns RiggedFinger scripts */
+    private void assignRiggedFingersByName(){
+      List<string> palmStrings = new List<string> { "palm", "Palm", "PALM" };
```

Comment:
One tricks you can use if you want.  I like to convert everything to lowercase before doing a comparison, that way you don't have to have a bunch of different versions of the same word.